src/moov/trak/mdia/minf/stbl/stsd/mp4a/mod.rs

```diff
@@ -7,49 +7,17 @@ pub use esds::Esds;
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Mp4a {
-    pub data_reference_index: u16,
-    pub channelcount: u16,
-    pub samplesize: u16,
-    pub samplerate: FixedPoint<u16>,
-    pub esds: Option<Esds>,
+    pub audio: Audio,
+    pub esds: Esds,
```

Comment:
@bradh Do you know if `Esds` is optional? I don't think it is based on looking at the specs.

And breaking change alert. `Audio` is `AudioSampleEntry` which is also used by Opus (and others).

Reply:
I think that ISO/IEC 14496-14:2020 Section 6.7.2 requires it for every `mp4a` sample entry.

That said, I'm not very familiar with the audio parts (or trying to work out what ISO 14496-12 is telling me, given the "yeah, nah" relationship with the systems (ISO 14496-1) work. So I checked the file format samples in https://github.com/MPEGGroup/FileFormatConformance/tree/main/data/file_features and every `mp4a` file has an `esds`.

(Only because you flagged `AudioSampleEntry` here:) I don't see `esds` as a required box in a generic `AudioSampleEntry`, only in the `mp4a` specialisation.